Source: https://github.com/google/googletest/commit/61f010d703b32de9bfb20ab90ece38ab2f25977f

commit 61f010d703b32de9bfb20ab90ece38ab2f25977f
Author: Abseil Team <absl-team@google.com>
Date:   Mon Mar 30 14:26:22 2020 -0400

    Googletest export
    
    Do not use std::result_of as it was removed in C++20.
    
    PiperOrigin-RevId: 303783600

diff --git a/googlemock/include/gmock/gmock-actions.h b/googlemock/include/gmock/gmock-actions.h
index b4127e93..615651b3 100644
--- a/googlemock/include/gmock/gmock-actions.h
+++ b/googlemock/include/gmock/gmock-actions.h
@@ -970,7 +970,8 @@ struct InvokeMethodWithoutArgsAction {
   Class* const obj_ptr;
   const MethodPtr method_ptr;
 
-  using ReturnType = typename std::result_of<MethodPtr(Class*)>::type;
+  using ReturnType =
+      decltype((std::declval<Class*>()->*std::declval<MethodPtr>())());
 
   template <typename... Args>
   ReturnType operator()(const Args&...) const {
